### PR TITLE
Feature/cid 2908 send multi manifest files

### PR DESCRIPTION
**Pull request by henriq-amaral-leanix**
## 🛠 Changes made
In order to support discovering multiple manifest files within a repository, this PR sends the files to the backend

## ✨ Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## 🧪 How Has This Been Tested?
- [x] Test adapted
- [x] Test added

## 🏎 Checklist:
- [x] My code follows the style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have made corresponding changes to the documentation (README.md)
- [x] My commit message clearly reflects the changes made
- [x] Assigned the appropriate labels (version, PR type, etc.)

### CODE REVIEW

File: src/main/kotlin/net/leanix/githubagent/dto/GitHubSearchResponse.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class ItemResponse(
+    val name: String,
+    val path: String,
+    val repository: RepositoryItemResponse,
+    val url: String,
+)
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class RepositoryItemResponse(
```
Comment (alfredo-mfaria) [inline src/main/kotlin/net/leanix/githubagent/dto/GitHubSearchResponse.kt:22]: Don't we want the url as well?

File: src/main/kotlin/net/leanix/githubagent/dto/ManifestFilesDTO.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package net.leanix.githubagent.dto
+
+data class ManifestFilesDTO(
+    val repositoryId: String,
+    val repositoryFullName: String,
+    val manifestFiles: List<ManifestFileDTO>
+)
+
+data class ManifestFileDTO(
+    val path: String,
```
Comment (alfredo-mfaria) [inline src/main/kotlin/net/leanix/githubagent/dto/ManifestFilesDTO.kt:10]: is it the para or the complete url?

File: src/main/kotlin/net/leanix/githubagent/dto/ManifestFilesDTO.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    val content: String?,
```
Comment (alfredo-mfaria) [inline src/main/kotlin/net/leanix/githubagent/dto/ManifestFilesDTO.kt:11]: Should the content be mantadory? an empty file would be an empty String. Not sure about this

File: src/main/kotlin/net/leanix/githubagent/dto/ManifestFilesDTO.kt (hunk 2 above)
Comment (henriq-amaral-leanix) [reply src/main/kotlin/net/leanix/githubagent/dto/ManifestFilesDTO.kt:10]: Today, the path is enough to mount the path to the file on the backend.
Repository url(already persisted on the backend) + path

File: src/main/kotlin/net/leanix/githubagent/dto/ManifestFilesDTO.kt (hunk 3 above)
Comment (henriq-amaral-leanix) [reply src/main/kotlin/net/leanix/githubagent/dto/ManifestFilesDTO.kt:11]: good point, I'll fix it

**Review by alfredo-mfaria — APPROVED** [review]

**Pull request merged**